app/dashboard/date-picker/step-btn.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCaretDown } from '@fortawesome/free-solid-svg-icons/faCaretDown';
+import { twentyFourHoursInMilliseconds } from '@/app/utils/constants';
+
+interface StepBtnProps {
+  increment: boolean;
+  startDate: Date;
+  minDate: Date;
+  maxDate: Date;
+  onChange: (date: Date | null) => void;
+}
+
+const StepBtn = ({
+  increment,
+  startDate,
+  minDate,
+  maxDate,
+  onChange,
+}: StepBtnProps) => {
+  const isDateOutOfBound = (date: Date): boolean => {
+    const dateInDay = new Date(date.toDateString()).getTime();
+    const minDateInDay = new Date(minDate.toDateString()).getTime();
+    const maxDateInDay = new Date(maxDate.toDateString()).getTime();
+
+    return dateInDay < minDateInDay || dateInDay > maxDateInDay;
```

Comment:
Optional: wrap this function in `useCallback` to memoize it so that it won't have to redefine the function on every render.